src/Generator/Mixin.php

```diff
@@ -0,0 +1,187 @@
+
+            $this->createNullOr($method, $supportedMethods, $mixin);
+            $this->createAll($method, $supportedMethods, $mixin);
+        }
+
+        $printer = new PsrPrinter();
+
+        return $printer->printFile($file);
+    }
+
+    private function parseDocComment(string $comment): array
+    {
+        $lines = explode("\n", $comment);
+
+        $result = [];
+
+        foreach ($lines as $line) {
+            if (preg_match('~^\*\s+@(\S+)(\s+.*)?$~', trim($line), $matches)) {
+                if (count($matches) === 2 || $matches[2] === null) {
+                    $matches[2] = '';
+                }
+                $result[$matches[1]][] = trim($matches[2]);
+            }
+        }
+
+        return $result;
+    }
+
+    private function splitDocLine(string $line): array
+    {
+        if (!preg_match('~^(.*)\s+(\$\S+)$~', $line, $matches)) {
+            return [$line];
+        }
+
+        return [trim($matches[1]), $matches[2]];
+    }
+
+    private function createWrapper(ReflectionMethod $method, array $supportedMethods, ClassType $mixin, $methodNameTemplate, $typeTemplate)
+    {
+        $newMethodName = sprintf($methodNameTemplate, ucfirst($method->name));
+
+        if (!in_array($newMethodName, $supportedMethods, true)) {
+            return;
+        }
+
+        $comment = $method->getDocComment();
+
+        $parsedComment = $this->parseDocComment($comment);
+
+        if (!in_array('psalm-assert', array_keys($parsedComment))) {
+            return;
+        }
+
+        $mixinMethod = $mixin->addMethod($newMethodName)
+            ->setStatic();
+
+        $parameters = $method->getParameters();
+
+        foreach ($parameters as $parameter) {
+            $mixinParameter = $mixinMethod->addParameter($parameter->name);
+
+            if ($parameter->isDefaultValueAvailable()) {
+                $mixinParameter->setDefaultValue($parameter->getDefaultValue());
+            }
+        }
+
+        foreach ($parsedComment as $key => $values) {
+            foreach ($values as $value) {
+                $parts = $this->splitDocLine($value);
+                $type = $parts[0];
+
+                if ($key === 'psalm-assert') {
+                    $type = sprintf($typeTemplate, $type);
+                }
+
+                $comment = sprintf('@%s %s', $key, $type);
+                if (count($parts) >= 2) {
+                    $comment .= sprintf(' %s', implode(' ', array_slice($parts, 1)));
+                }
+
+                $mixinMethod->addComment($comment);
+            }
+        }
+    }
+
+    private function createNullOr(ReflectionMethod $method, array $supportedMethods, ClassType $mixin)
+    {
+        $this->createWrapper($method, $supportedMethods, $mixin, 'nullOr%s', 'null|%s');
+    }
+
+    private function createAll(ReflectionMethod $method, array $supportedMethods, ClassType $mixin)
+    {
+        $this->createWrapper($method, $supportedMethods, $mixin, 'all%s', 'array<array-key,%s>');
```

Comment:
The first param of the `all*` methods does not need to be of type `array`. any iterable is fine.